contracts/Factory.sol

```diff
@@ -1,44 +1,42 @@
 pragma solidity ^0.4.18;
 
-import './IRegistry.sol';
-import '../upgradeability/OwnedUpgradeabilityProxy.sol';
+import './upgradeability/Proxy.sol';
+import './upgradeability/OwnedUpgradeabilityProxy.sol';
+import './Registry.sol';
 
 
 /**
- * @title Registry
- * @dev This contract works as a registry of versions, it holds the implementations for the registered versions.
+ * @title Factory
+ * @dev 
  */
-contract Registry is IRegistry {
-  // Mapping of versions to implementations of different functions
-  mapping (string => address) private versions;
+contract Factory {
+
+  Registry public _registry;
+
+  function Factory(Registry registry) public {
+    _registry = registry;
+  }
 
   /**
-  * @dev Registers a new version with its implementation address
-  * @param version representing the version name of the new implementation to be registered
-  * @param implementation representing the address of the new implementation to be registered
+  * @dev Tells the address of registry
+  * @return address of the registry
   */
-  function addVersion(string version, address implementation) public {
-    require(versions[version] == address(0));
-    require(implementation != address(0));
-    versions[version] = implementation;
-    VersionAdded(version, implementation);
+  function registry() public view returns (Registry) {
+    return _registry;
   }
 
   /**
-  * @dev Tells the address of the implementation for a given version
-  * @param version to query the implementation of
-  * @return address of the implementation registered for the given version
+  * @dev This event will be emitted every time a new proxy is created
+  * @param proxy representing the address of the proxy created
   */
-  function getVersion(string version) public view returns (address) {
-    return versions[version];
-  }
+  event ProxyCreated(address proxy);
 
   /**
   * @dev Creates an upgradeable proxy upgraded to an initial version
   * @param version representing the first version to be set for the proxy
   * @return address of the new proxy created
   */
-  function createProxy(string version) public returns (OwnedUpgradeabilityProxy) {
+  function createProxy(string version) public returns (Proxy) {
```

Comment:
Why not returning an `OwnedUpgradeabilityProxy` directly, since we don't need to conform to any interface that returns just `Proxy`? Would this break the compilation?